libs/stats/odc/stats/_cli_generate_mosaic.py

```diff
@@ -0,0 +1,125 @@
+from datacube.utils.rio import configure_s3_access
+
+
+def bytes2ds_doc(data):
+    if isinstance(data, bytes):
+        data = data.decode("utf-8")
+    stac_doc = json.loads(data)
+    eo3_doc = stac_transform(stac_doc)
+    ds_doc = prep_eo3(eo3_doc)
+    return ds_doc
+
+
+def blob2ds(blob, product):
+    doc = bytes2ds_doc(blob.data)
+    return Dataset(product, doc, uris=[blob.url])
+
+
+def s3_fetch_dss(base, product, glob="*.json", s3=None):
+    if s3 is None:
+        s3 = S3Fetcher(aws_unsigned=True)
+    blobs = s3(o.url for o in s3.find(base, glob=glob))
+    dss = (blob2ds(b, product) for b in blobs)
+    return dss
+
+
+def xr_to_mem(xx, client):
+    data = store_to_mem(xx.data, client)
+    return xr.DataArray(
+        data=data, coords=xx.coords, dims=xx.dims, attrs=xx.attrs
+    )
+
+
+def save(xx, location, product_name, verbose):
+    client = start_local_dask(
+        nanny=False,
+        n_workers=1,
+        threads_per_worker=8,
+        mem_safety_margin="0G",
+        processes=False,
+    )
+
+    gdal_cfg = {"GDAL_CACHEMAX": 8 * (1 << 30)}
+    configure_s3_access(aws_unsigned=True, cloud_defaults=True, **gdal_cfg)
+    configure_s3_access(
+        aws_unsigned=True, cloud_defaults=True, client=client, **gdal_cfg
+    )
+
+    rgba = to_rgba(xx.isel(time=0), clamp=(0, 3000))
+    rgba = xr_to_mem(rgba, client)
+
+    if verbose:
+        print(f"Writing {location}/{product_name}.tif")
+
+    write_cog(
+        rgba,
+        f"{location}/{product_name}.tif",
+        blocksize=1024,
+        compress="zstd",
+        zstd_level=4,
+        overview_levels=[],
+        NUM_THREADS="ALL_CPUS",
+        BIGTIFF="YES",
+        SPARSE_OK=True,
+    )
+
+
+@click.command()
+@click.argument("product", type=str)
+@click.argument("input_prefix", type=str)
+@click.argument("location", type=str)
+@click.option("--verbose", "-v", is_flag=True, help="Be verbose")
+def cli(product, input_prefix, location, verbose):
+    """
+    Generate mosaic overviews of the stats data.
+
+    An intermediate cache file is generated and stored in the output location
+    during this process.
+    Note: The input bucket must be public otherwise the data can not be listed.
+    """
+
+    product = product_from_yaml(product)
+    if verbose:
+        print(f"Preparing mosaics for {product.name} product")
+
+    dss = s3_fetch_dss(input_prefix, product, glob="*.json")
+    cache = create_cache(f"{product.name}.db")
+
+    if verbose:
+        print(f"Writing {location}/{product.name}.db")
+
+    cache = create_cache(f"{location}/{product.name}.db")
+    cache.bulk_save(dss)
+    if verbose:
+        print(f"Found {cache.count:,d} datasets")
+
+    dc = Datacube()
+    dss = list(cache.get_all())
+    xx = dc.load(
+        datasets=dss,
+        dask_chunks={"x": 3200, "y": 3200},
+        resolution=(-120, 120),
+        measurements=["red", "green", "blue"],
+    )
+
+    save(xx, location, product.name, verbose)
```

Comment:
We don't need this part in here.

Let's just make one tool for capturing db file, and then separate one for mosaic generation which will have to be way more configurable than this.

For this PR, just db saving please.